src/BSP/timer_if.c

```diff
@@ -223,22 +253,15 @@ UTIL_TIMER_Status_t TIMER_IF_StartTimer(uint32_t timeout)
   /* USER CODE BEGIN TIMER_IF_StartTimer */
 
   /* USER CODE END TIMER_IF_StartTimer */
-  RTC_AlarmTypeDef sAlarm = {0};
   /*Stop timer if one is already started*/
   TIMER_IF_StopTimer();
   timeout += RtcTimerContext;
 
   TIMER_IF_DBG_PRINTF("Start timer: time=%d, alarm=%d\n\r",  GetTimerTicks(), timeout);
-  /* starts timer*/
-  sAlarm.BinaryAutoClr = RTC_ALARMSUBSECONDBIN_AUTOCLR_NO;
-  sAlarm.AlarmTime.SubSeconds = UINT32_MAX - timeout;
-  sAlarm.AlarmMask = RTC_ALARMMASK_NONE;
-  sAlarm.AlarmSubSecondMask = RTC_ALARMSUBSECONDBINMASK_NONE;
-  sAlarm.Alarm = RTC_ALARM_A;
-  if (HAL_RTC_SetAlarm_IT(&hrtc, &sAlarm, RTC_FORMAT_BCD) != HAL_OK)
-  {
-    Error_Handler();
-  }
+
+  /* Program ALARM A on subsecond, mask is 32 (and fixed to RTC_ALARMMASK_NONE for calendar) */
+  RTC_StartAlarm(RTC_ALARM_A, 0, 0, 0, 0, UINT32_MAX - timeout, RTC_HOURFORMAT12_PM, 32UL);
```

Comment:
why RTC_HOURFORMAT12_PM ? 
below it is set to 
rtc.begin(true, STM32RTC::HOUR_24);

Reply:
Use the RTC handle as a global from the STM32_RTC library 
Use Alarm B of the RTC to trig on SubSeconds